File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr) {
    vector<vector<int>> answer;
    if(arr.size() > arr[0].size()){
        for(int i = 0; i < arr.size(); i++){
            arr[i].resize(arr.size());
        }
    } else {
        arr.resize(arr[0].size(), vector<int>(arr[0].size()));
    }
    return arr;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    const n = Math.max(arr.length, arr[0].length);
    const result = arr.map(r => [...r, ...Array(n - arr[0].length).fill(0)]);

    for(let i = arr.length; i < n; i++){
        result.push(Array(n).fill(0));
    }

    return result;
}
